osu.Game/Rulesets/Ruleset.cs

```diff
@@ -19,9 +21,11 @@ public abstract class Ruleset
 
         public virtual IEnumerable<BeatmapStatistic> GetBeatmapStatistics(WorkingBeatmap beatmap) => new BeatmapStatistic[] { };
 
+        public IEnumerable<Mod> GetAllMods() => Enum.GetValues(typeof(ModType)).Cast<ModType>().SelectMany(type => GetModsFor(type).Where(mod => mod != null).SelectMany(mod => (mod as MultiMod)?.Mods ?? new[] { mod }));

```

Comment:
Can you split this up into multiple lines at the dots (1 linq call per line) and add a comment for every line explaining what it does? Right now this takes quiiiite a while to mentally parse and understand.